camus-etl-kafka/src/test/java/com/linkedin/camus/etl/kafka/CamusJobTestWithMock.java

```diff
@@ -0,0 +1,383 @@
+  
+  private static final String KAFKA_HOST = "localhost";
+  private static final int KAFKA_PORT = 2121;
+  private static final int KAFKA_TIMEOUT_VALUE = 1000;
+  private static final int KAFKA_BUFFER_SIZE = 1024;
+  private static final String KAFKA_CLIENT_ID = "Camus";
+
+  private static final String TOPIC_1 = "topic_1";
+  private static final int PARTITION_1_ID = 0;
+  private static final int EARLY_OFFSET = 0;
+  private static final int LATE_OFFSET = 1;
+
+  private static FileSystem fs;
+  private static Gson gson;
+  private static Map<String, List<MyMessage>> messagesWritten;
+
+  // mock objects
+  private static List<Object> _mocks = new ArrayList<Object>();
+  private static SimpleConsumer _simpleConsumer;
+
+  @BeforeClass
+  public static void beforeClass() throws IOException {
+    fs = FileSystem.get(new Configuration());
+    gson = new Gson();
+
+    // You can't delete messages in Kafka so just writing a set of known messages that can be used for testing
+    messagesWritten = new HashMap<String, List<MyMessage>>();
+    messagesWritten.put(TOPIC_1, writeKafka(TOPIC_1, 1));
+  }
+
+  @AfterClass
+  public static void afterClass() {
+  }
+
+  private Properties props;
+  private CamusJob job;
+  private TemporaryFolder folder;
+  private String destinationPath;
+
+  @Before
+  public void before() throws IOException, NoSuchFieldException, IllegalAccessException {
+    resetCamus();
+
+    folder = new TemporaryFolder();
+    folder.create();
+
+    String path = folder.getRoot().getAbsolutePath();
+    destinationPath = path + DESTINATION_PATH;
+
+    props = new Properties();
+
+    props.setProperty(EtlMultiOutputFormat.ETL_DESTINATION_PATH, destinationPath);
+    props.setProperty(CamusJob.ETL_EXECUTION_BASE_PATH, path + EXECUTION_BASE_PATH);
+    props.setProperty(CamusJob.ETL_EXECUTION_HISTORY_PATH, path + EXECUTION_HISTORY_PATH);
+
+    props.setProperty(EtlInputFormat.CAMUS_MESSAGE_DECODER_CLASS, JsonStringMessageDecoder.class.getName());
+    props.setProperty(EtlMultiOutputFormat.ETL_RECORD_WRITER_PROVIDER_CLASS,
+        SequenceFileRecordWriterProvider.class.getName());
+
+    props.setProperty(EtlMultiOutputFormat.ETL_RUN_TRACKING_POST, Boolean.toString(false));
+    props.setProperty(CamusJob.KAFKA_CLIENT_NAME, KAFKA_CLIENT_ID);
+    props.setProperty(CamusJob.KAFKA_TIMEOUT_VALUE, Integer.toString(KAFKA_TIMEOUT_VALUE));
+    props.setProperty(CamusJob.KAFKA_FETCH_BUFFER_SIZE, Integer.toString(KAFKA_BUFFER_SIZE));
+
+    props.setProperty(CamusJob.KAFKA_BROKERS, KAFKA_HOST + ":" + KAFKA_PORT);
+
+    // Run Map/Reduce tests in process for hadoop2  
+    props.setProperty("mapreduce.framework.name", "local");
+    // Run M/R for Hadoop1
+    props.setProperty("mapreduce.jobtracker.address", "local");
+
+    job = new CamusJob(props);
+  }
+
+  @After
+  public void after() throws IOException {
+    EasyMock.verify(_mocks.toArray());
```

Comment:
Is calling verify() in After method a good practice? What if there's a test that doesn't need to verify all these mocks?